### PR TITLE
Add workflow for building and uploading PDF.

### PR DESCRIPTION
**Pull request by Zimmi48**
This will automatically build and deploy the PDF every time it is updated:

- When a commit is pushed to `master`, the PDF will be deployed to GitHub Pages, allowing to view it online at https://coq-community.org/hydra-battles/hydras.pdf (preview at https://Zimmi48.github.io/hydra-battles/hydras.pdf).

- When a commit is pushed to another branch or a pull request, the PDF will be deployed as an artifact which can be downloaded and unzipped for reviewing (preview at https://github.com/Zimmi48/hydra-battles/actions/runs/318159714).

### CODE REVIEW

File: .github/workflows/build-pdf.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: actions/checkout@v2
+      - name: Compile LaTeX document
+        uses: xu-cheng/latex-action@v2
+        with:
+          root_file: hydras.tex
+          working_directory: doc
+          latexmk_use_lualatex: true
+      - name: Extract PDF
+        run: |
+          mkdir public
```
Comment (palmskog) [inline .github/workflows/build-pdf.yml:17]: shouldn't this be `mkdir -p` if you want to avoid errors when the directory exists?

Comment (Zimmi48) [reply .github/workflows/build-pdf.yml:17]: Yeah, why not make it future-proof indeed!

Comment (Zimmi48) [reply .github/workflows/build-pdf.yml:17]:
```suggestion
          mkdir -p public
```

**Comment by palmskog** [comment]
One issue to keep in mind is that each PDF is around 1MB. Git does have delta-compression, but with hundreds of commits, repo size could become a factor.

**Comment by anton-trunov** [comment]
@palmskog Does this workflow actually commit build artifacts? I would expect it to upload them to an external storage.

**Comment by palmskog** [comment]
> Does this workflow actually commit build artifacts? I would expect it to upload them to an external storage.

I based my comment on the following:
> the PDF will be deployed to GitHub Pages

**Comment by anton-trunov** [comment]
I see, thanks. I'm wondering if we could just link latest artifact from Github Pages.

**Comment by Zimmi48** [comment]
> One issue to keep in mind is that each PDF is around 1MB. Git does have delta-compression, but with hundreds of commits, repo size could become a factor.

This is avoided by the fact that the gh-pages branch is erased / force-pushed at every deploy. It never contains more than one commit.

For the record, my initial plan was indeed to push an artifact and not a GitHub Pages branch, but GitHub Actions won't allow viewing the PDF in the browser in that case (it encapsulates it in a zip archive). GitLab CI is superior here, but using it would introduce additional complexity (not worth it IMHO).

**Pull request merged**